module/VuFind/tests/unit-tests/src/VuFindTest/Form/FormTest.php

```diff
@@ -564,4 +566,36 @@ public function testCheckboxRequiredValidators()
             $this->assertFalse($form->isValid());
         }
     }
+
+    /**
+     * Test email subjects.
+     *
+     * @return void
+     */
+    public function testEmailSubjects()
```

Comment:
This would be a really good place to use a `@dataProvider` annotation -- instead of hard-coding $ids and $results into the body of the function, you can create a separate function that returns an array of test data (keyed by descriptive text), and it will get sent to the test method one set at a time. It makes everything a bit cleaner and easier to extend.  It would look something like:

<pre>
public function getEmailSubjectsData()
{
    return [
         'with placeholders' => ['TestSubjectEmailWithPlaceHolders', 'Subject One Two option-1'],
         'without placeholders' => ['TestSubjectEmailWithoutPlaceholders', 'Subject without placeholders'],
    ];
}

/**
 * @dataProvider getEmailSubjectsData
 */
public function testEmailSubjects($formToTest, $expectedSubject)
{
   ...
</pre>

... and then you just use the function parameters instead of having to do a for loop and index through the results. Does that make sense?

Reply:
Oh, that makes actually sense, will try to adjust it tomorrow!

Reply:
Looks great -- thanks! I only learned about dataProviders recently, but they're really useful! :-)